schematic/store/synapse.py

```diff
@@ -1713,8 +1753,47 @@ def add_annotations_to_entities_files(
 
             # Adding annotations to connected files.
             if entityId:
-                self._add_annotations(dmge, row, entityId, hideBlanks, annotation_keys)
-                logger.info(f"Added annotations to entity: {entityId}")
+                # Format annotations for Synapse
+                annos_task = asyncio.create_task(
+                    self.format_row_annotations(
+                        dmge, row, entityId, hideBlanks, annotation_keys
+                    )
+                )
+                requests.add(annos_task)
+
+                while requests:
+                    done_tasks, pending_tasks = await asyncio.wait(
+                        requests, return_when=asyncio.FIRST_COMPLETED
+                    )
+                    requests = pending_tasks
+
+                for completed_task in done_tasks:
+                    try:
+                        annos = completed_task.result()
+
+                        if isinstance(annos, Annotations):
+                            annos_dict = asdict(annos)
+                            entity_id = annos_dict["id"]
+                            logger.info(
+                                f"Successfully stored annotations for {entity_id}"
+                            )
+                        else:
+                            # remove special characters in annotations
+                            entity_id = annos["EntityId"]
+                            logger.info(
+                                f"Obtained and processed annotations for {entity_id} entity"
+                            )
+                            if annos:
+                                requests.add(
+                                    asyncio.create_task(
+                                        self.store_async_annotation(
+                                            annotation_dict=annos
+                                        )
+                                    )
+                                )
+
+                    except Exception as e:
+                        raise RuntimeError(f"failed with { repr(e) }.")
```

Comment:
I think the syntax you would want here would be something like
`raise RuntimeError("message") from e`

https://stackoverflow.com/questions/24752395/python-raise-from-usage

Reply:
Thanks for your help! I already addressed that :)